variables/collection.go

```diff
@@ -131,8 +132,9 @@ func (c *Collection) SortedOverridingVariables() []string {
 	return c.overriding().SortedAny()
 }
 
-// Expand expands variables within the given word.
-func (c *Collection) Expand(word string) string {
+// ExpandOld expands variables within the given word, it does not peform shelling-out.
+// it will eventually be removed when the ShellOutAnywhere feature is fully-adopted
+func (c *Collection) ExpandOld(word string) string {
```

Comment:
Nit: I noticed that `ExpandOld` is exactly the same as Expand, but with `shellOut` set to `nil`. Perhaps this could be removed altogether and just set nil from the caller?

Reply:
it's actually quite different: `dfShell.NewLex('\\')` vs `shell.NewLex('\\')` (the original buildkit code vs our forked version).

It's theoretically possible to write an interface for them, but I think that's overkill.

Reply:
:+1: